client/app/visualizations/table/utils.js

```diff
@@ -184,7 +192,8 @@ export function sortRows(rows, orderBy) {
     for (let i = 0; i < orderBy.length; i += 1) {
       va = a[orderBy[i].name];
       vb = b[orderBy[i].name];
-      if (isNil(va) || va < vb) {
+
+      if (chechNumber(va) && chechNumber(vb)) {
```

Comment:
1. `chechNumber` does not cover all possible cases (e.g. it will return `false` for negative numbers. It is better to use something line `isFinite(parseFloat(value))` (also function name has a typo, and it does not reflect the purpose of the function).
2. this line (which I'm adding comment to) totally breaks reverse sorting - it disables `isNil` check, actual value comparison is missing. Also, there is similar check few lines below and it's not updated.
3. this fix in whole is very specific - it will try to handle all values as numbers. But what about other types? Strings, date/time, etc.? The proper fix should be based on idea from getredash/redash#5111 - sorting should be performed according to column type selected in table editor. E.g. if query returns strings, but user chose that column to be displayed as number - sort function should cast all values to number; same for date/time and other types.

Reply:
Sorry, i should change `re` to `const re = /^-?\d*\.?\d+$/;`, this will cover all possible number. I found only float number can't sort correctly, this bug because of treat float number as string, this PR will not affect other type(string, date).

Reply:
@kravets-levko can you check this re? thank you!

Reply:
@yankeeguyu The regex still does not cover all cases, e.g. `+314e-2` is still a number, but it won't be recognized. Please check my first comment in this thread. I already offered a variant that will work for sure in item 1. Also, items 2 and 3 from my reply are still relevant.

Reply:
> @yankeeguyu The regex still does not cover all cases, e.g. `+314e-2` is still a number, but it won't be recognized. Please check my first comment in this thread. I already offered a variant that will work for sure in item 1. Also, items 2 and 3 from my reply are still relevant.

Thank you very much!

Reply:
> @yankeeguyu The regex still does not cover all cases, e.g. `+314e-2` is still a number, but it won't be recognized. Please check my first comment in this thread. I already offered a variant that will work for sure in item 1. Also, items 2 and 3 from my reply are still relevant.

Although the current PR can solve my problem, i still want to fix this bug completely. This bug can be fixed?
I tried some modifications, but the type of number is still parsed as string at this problem. Thank you!

Reply:
> @yankeeguyu The regex still does not cover all cases, e.g. `+314e-2` is still a number, but it won't be recognized. Please check my first comment in this thread. I already offered a variant that will work for sure in item 1. Also, items 2 and 3 from my reply are still relevant.

Hi this function i change it to 
`
function checkNumber(n) {
     return Number(n) === n;
}
`

<img width="300" alt="image" src="https://user-images.githubusercontent.com/46274612/205228912-af8ba411-55a9-48b0-b06e-4e3143a5ca0d.png">

Reply:
> > @yankeeguyu The regex still does not cover all cases, e.g. `+314e-2` is still a number, but it won't be recognized. Please check my first comment in this thread. I already offered a variant that will work for sure in item 1. Also, items 2 and 3 from my reply are still relevant.
> 
> Hi this function i change it to `function checkNumber(n) { return Number(n) === n; }`
> 
> <img alt="image" width="300" src="https://user-images.githubusercontent.com/46274612/205228912-af8ba411-55a9-48b0-b06e-4e3143a5ca0d.png">

Sorry. This has bug.